src/CommonITILTask.php

```diff
@@ -291,10 +291,55 @@ private function handleTaskDuration(array &$input, int $timestart, int $timeend)
         }
     }
 
+    public function assignTechFromtask($input)
+    {
+        //if user or group assigned to ticket task, add it to the ticket
+        $entities = [
+            'Ticket' => [
+                'users_id_tech' => new Ticket_User(),
+                'groups_id_tech' => new Group_Ticket()
+            ],
+            'Change' => [
+                'users_id_tech' => new Change_User(),
+                'groups_id_tech' => new Change_Group()
+            ],
+            'Problem' => [
+                'users_id_tech' => new Problem_User(),
+                'groups_id_tech' => new Group_Problem()
+            ],
+        ];
+        foreach ($entities as $entityType => $entityData) {
+            $entityType = strtolower($entityType);
+            if (isset($input[$entityType . 's_id'])) {
+                foreach ($entityData as $key => $entity) {
+                    if (isset($input[$key]) && !empty($input[$key])) {
+                        $entityId = str_replace('_tech', '', $key);
+                        if (
+                            $entity->getFromDBByCrit(
+                                [
+                                    $entityType . 's_id' => $input[$entityType . 's_id'],
+                                    $entityId => $input[$key],
+                                    'type' => 2
```

Comment:
The constant for the actor types are found in the `CommonITILActor` class.

For example you can use `CommonITILActor::ASSIGN` here instead of 2 :)